cli/src/cli_util.rs

```diff
@@ -1689,6 +1689,13 @@ Then run `jj squash` to move the resolution into the conflicted commit."#,
     }
 }
 
+/// A [`Transaction`] tied to a particular workspace.
+/// `WorkspaceCommandTransaction`s are created with
+/// [`WorkspaceCommandHelper::start_transaction`] and committed with
+/// [`WorkspaceCommandTransaction::finish`]. The inner `Transaction` can also be
+/// extracted using [`WorkspaceCommandTransaction::into_inner`] and committed
+/// with [`WorkspaceCommandHelper::finish_transaction`] in situations where
+/// access to the inner `Transaction` is necessary.
```

Comment:
nit: `.finish_transaction()` is private, so the external callers will have to guarantee consistency by themselves.

Reply:
That's a good point. Maybe `into_inner()` should also be private? It seems like it's use outside of cli_util.rs could be replaced by a call to `start_repo_transaction`. It only have two external callers, `create_and_check_out_recovery_commit` and `cmd_untrack`.

I'm not sure exactly why those callers need the bare transaction instead of using it through the workspace. I'm going to take a closer look, but maybe you know? It would be good to document it.

Reply:
> Maybe `into_inner()` should also be private? It seems like it's use outside of cli_util.rs could be replaced by a call to `start_repo_transaction`.

`start_repo_transaction()` would have the same problem as `into_inner()`. Callers need to aware that the workspace isn't managed by the inner transaction, and the workspace command helper is kinda detached from the transaction. I think the name `.into_inner()` suggests that it's low-level API, so maybe it's okay?

> I'm not sure exactly why those callers need the bare transaction instead of using it through the workspace.

iirc, they do need to update the workspace in unusual way.

Reply:
> iirc, they do need to update the workspace in unusual way.

Yeah, I haven't grok'd the details but that's what it looks like.

> start_repo_transaction() would have the same problem as into_inner()

Yes, definitely. I was just thinking maybe we don't need both, but replacing a call to `into_inner` with `start_repo_transaction` requires input args from both the command helper and the command, which might be a little confusing.

```rust
// This:    
let mut tx = workspace_command.start_transaction().into_inner();

// Becomes this:    
let mut tx = start_repo_transaction(
    workspace_command.repo(),
    command.settings(),
    command.string_args(),
);
```

Let's just leave it as-is for now. I'm going to update the doc comments.